### PR TITLE
Fix-2570 Updated repo path to Go module

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/#2570
Review deadline: 3 July

## Page previews

[Preview of Build section of Lambda guide](https://deploy-preview-17--redpanda-connect.netlify.app/redpanda-connect/guides/serverless/lambda/#build)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [x] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by Jeffail — APPROVED** [review]
LGTM 👍

File: modules/guides/pages/serverless/lambda.adoc [hunk 1]
```diff
@@ -163,6 +163,6 @@ You can build and archive the function yourself with:
 
 [source,sh]
 ----
-go build github.com/benthosdev/benthos/v4/cmd/serverless/benthos-lambda
+go build github.com/redpanda-data/connect/v4/cmd/serverless/connect-lambda
 zip benthos-lambda.zip benthos-lambda
```
Comment (JakeSCahill) [inline modules/guides/pages/serverless/lambda.adoc:167]: Would this be `connect-lambda`?

Comment (asimms41) [reply modules/guides/pages/serverless/lambda.adoc:167]: @JakeSCahill. I checked with Ash but also here is the repo path: [https://github.com/redpanda-data/connect/blob/main/cmd/serverless/connect-lambda/main.go](https://github.com/redpanda-data/connect/blob/main/cmd/serverless/connect-lambda/main.go)

Comment (asimms41) [reply modules/guides/pages/serverless/lambda.adoc]: Checked and updated. Good spot!

**Review by Jeffail — APPROVED** [review]
LGTM 👍

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**